File: src/tool/ReadToolView.ts
class ReadToolView extends SecondarySessionPartView {
  constructor(tool: Tool, openFile: (path: string) => void, parts: ToolParts) {
    super(parts.header, parts.scroll(tool), false);
    this.setup(tool, parts, openFile);
  }

  private setup(tool: Tool, parts: ToolParts, openFile: (path: string) => void): void {
    // Setup logic here
  }
}

File: src/tool/ReadToolView.kt
class ReadToolView(
    tool: Tool,
    openFile: SessionFileOpener = { _, _ -> },
    private val selection: SessionSelection? = null,
    private val parts: ToolParts = toolParts(tool, openFile),
) : SecondarySessionPartView(parts.header, parts.scroll(tool), expandable = false) {
}
